numpy/core/include/numpy/npy_math.h

```diff
@@ -111,6 +111,18 @@ NPY_INLINE static float __npy_nzerof(void)
 #define NPY_SQRT2l    1.414213562373095048801688724209698079L /* sqrt(2) */
 #define NPY_SQRT1_2l  0.707106781186547524400844362104849039L /* 1/sqrt(2) */
 
+/* Magic binary numbers used by popcount
+ * For type T, the magic numbers are computed as follows:
+ * Magic[0]: 0b01 01 01 01 01 01... = (T)~(T)0/3
+ * Magic[1]: 0b0011 0011 0011...    = (T)~(T)0/15  * 3
+ * Magic[2]: 0b00001111 00001111... = (T)~(T)0/255 * 15
+ * Magic[3]: 0b00000001 00000001... = (T)~(T)0/255
+ */
+static const npy_uint8  MAGIC8[]  = {0x55,               0x33,               0x0F,               0x01};
+static const npy_uint16 MAGIC16[] = {0x5555,             0x3333,             0x0F0F,             0x0101};
+static const npy_uint32 MAGIC32[] = {0x55555555,         0x33333333,         0x0F0F0F0F,         0x01010101};
+static const npy_uint64 MAGIC64[] = {0x5555555555555555, 0x3333333333333333, 0x0F0F0F0F0F0F0F0F, 0x0101010101010101};
```

Comment:
These look like they should be in `npy_math_internal.h.src` with the function that uses them?

Reply:
Fixed in 15dd4fa